eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTest.java

```diff
@@ -0,0 +1,325 @@
+package com.netflix.discovery.util;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.shared.Application;
+import com.netflix.discovery.shared.Applications;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class EurekaEntityFunctionsTest {
+
+    private InstanceInfo createSingleInstanceApp(
```

Comment:
This does not create an app but an instance.

Reply:
method name changed

Reply:
Apologies I misread that. Yes I agree that creating an `Application` is possible and better. However, `Mockito` [does not allow](https://github.com/mockito/mockito/wiki/FAQ) `hashcode()` to be mocked and since most `assertsEquals` compares the different `Instances` the tests fail to run. Which is why I created the method to return an `InstanceInfo` which can be added to an `Application` and then referenced from the `assertEquals`

Reply:
I updated the method to return an `Application`